plugins/extractors/bigquery/bigquery.go

```diff
@@ -181,10 +174,20 @@ func (e *Extractor) extractTable(ctx context.Context, ds *bigquery.Dataset, emit
 
 // Build the bigquery table metadata
 func (e *Extractor) buildTable(ctx context.Context, t *bigquery.Table, md *bigquery.TableMetadata) *assetsv1beta1.Table {
+	var tableStats *auditlog.TableStats
+	if e.config.IsCollectTableUsage {
+		// Fetch and extract logs first to build a map
+		var errL error
+		tableStats, errL = e.galClient.Collect(ctx, t.TableID)
```

Comment:
CMIIW, so now we will fetch table usage once for each table, which will create more requests but we won't fetch unused data/rows? @mabdh

Reply:
@StewartJingga yes correct